### ISSUE TITLE
Non-fatal errors on CentOS6

### ISSUE THREAD

**Issue by 011121** [john-kurkowski/tldextract#114]
On CentOS6 I'm seeing some nonfatal errors pop up when using tldextract:

```
python3.5/site-packages/tldextract/tldextract.py:223: ResourceWarning: unclosed <ssl.SSLSocket fd=5, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('10.32.216.57', 59070), raddr=('63.245.213.24', 443)>

  raw_suffix_list_data = find_first_response(self.suffix_list_urls)

python3.5/site-packages/tldextract/tldextract.py:295: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead

  LOG.warn("unable to cache TLDs in file %s: %s", self.cache_file, ioe)

unable to cache TLDs in file python3.5/site-packages/tldextract/.tld_set: [Errno 13] Permission denied: 'python3.5/site-packages/tldextract/.tld_set'
```

I've redacted some information from the file paths above but nothing important, I believe.

**Comment by john-kurkowski** [comment]
I'm curious about the `ResourceWarning`. I can't tell what it is at a glance. Would sure like to squash it.

The `DeprecationWarning` is a no brainer how to fix. Good catch.

The `Permission denied` is indeed expected if you do not have write perms to that folder. See also [this comment](https://github.com/john-kurkowski/tldextract/issues/9#issuecomment-4644805). Maybe you installed tldextract with `sudo`? Can you make that folder writable by the same user you're tldextract running as? Or don't install it with `sudo` etc.

**Comment by john-kurkowski** [comment]
See also the caching section of the readme, to specify an alternate path to cache in.

**Comment by 011121** [comment]
Yeah, I wasn't too worried about the permission error, I just included it for completeness.

**Comment by john-kurkowski** [comment]
Deprecation was fixed in #104. Unclosed python-requests session fixed in #140.
